File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();

        int sum =0;
        for(int i =0;i<n;i++){
            sum+=nums[i];
        }

        int leftSum =0;
        int rightSum = sum;

        for(int i =0;i<n;i++){
            leftSum +=nums[i];

            if(leftSum == rightSum){
                return i;
            }
            rightSum -= nums[i];
        }
        return -1;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)

        sum =0
        for i in range(n):
            sum +=nums[i]
        
        leftSum =0
        rightSum =sum 

        for i in range(n):
            leftSum += nums[i]

            if leftSum == rightSum:
                return i
            
            rightSum -= nums[i]
        return -1